### PR TITLE
feature-15

### PR DESCRIPTION
**Pull request by samuskitchen**
## Summary
- Swagger documentation added in user
## Trello Card
https://trello.com/c/pekPZlti/40-swagger

### CODE REVIEW

File: src/main/java/wolox/training/controllers/UserController.java [hunk 1]
```diff
@@ -48,7 +60,14 @@ public UserController(UserRepository userRepository, BookRepository bookReposito
...
     @GetMapping("/username/{username}")
-    public User findByName(@PathVariable String username) {
+    @ApiOperation(value = "Giving an username, return one user", response = User.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Success"),
+            @ApiResponse(code = 401, message = "Not Authorized"),
+            @ApiResponse(code = 403, message = "Access forbidden"),
+            @ApiResponse(code = 404, message = "User Not Found")
+    })
+    public User findByUsername(@PathVariable String username) {
```
Comment (JaimLus) [inline src/main/java/wolox/training/controllers/UserController.java]: I would add @ApiParam() for the username parameter.

File: src/main/java/wolox/training/controllers/UserController.java [hunk 2]
```diff
@@ -59,6 +78,13 @@ public User findByName(@PathVariable String username) {
...
      */
     @GetMapping("/{id}")
+    @ApiOperation(value = "Giving an id, return one user", response = User.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Success"),
+            @ApiResponse(code = 401, message = "Not Authorized"),
+            @ApiResponse(code = 403, message = "Access forbidden"),
+            @ApiResponse(code = 404, message = "User Not Found")
+    })
     public User findOne(@PathVariable Long id) {
```
Comment (JaimLus) [inline src/main/java/wolox/training/controllers/UserController.java]: I would add @ApiParam() for the id parameter.

File: src/main/java/wolox/training/controllers/UserController.java [hunk 3]
```diff
@@ -81,6 +113,13 @@ public User create(@RequestBody User user) {
...
      */
     @DeleteMapping("/{id}")
+    @ApiOperation(value = "Deletes a user")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully deleted user"),
+            @ApiResponse(code = 400, message = "Bad request"),
+            @ApiResponse(code = 404, message = "Resource not found"),
+            @ApiResponse(code = 500, message = "Internal server error")
+    })
     public void delete(@PathVariable Long id) {
```
Comment (JaimLus) [inline src/main/java/wolox/training/controllers/UserController.java]: I would add @ApiParam() for the id parameter.

File: src/main/java/wolox/training/controllers/UserController.java [hunk 4]
```diff
@@ -94,7 +133,14 @@ public void delete(@PathVariable Long id) {
...
     @PutMapping("/{id}")
-    public User updateBook(@RequestBody User user, @PathVariable Long id) {
+    @ApiOperation(value = "Updates a user", response = User.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully updated user"),
+            @ApiResponse(code = 400, message = "Bad request"),
+            @ApiResponse(code = 404, message = "Resource not found"),
+            @ApiResponse(code = 500, message = "Internal server error")
+    })
+    public User updateUser(@RequestBody User user, @PathVariable Long id) {
```
Comment (JaimLus) [inline src/main/java/wolox/training/controllers/UserController.java]: I would add @ApiParam() for the id parameter.

File: src/main/java/wolox/training/controllers/UserController.java [hunk 5]
```diff
@@ -112,6 +158,13 @@ public User updateBook(@RequestBody User user, @PathVariable Long id) {
...
     @PostMapping("/{userId}/books/{bookId}")
     @ResponseStatus(HttpStatus.CREATED)
+    @ApiOperation(value = "Given the id of a user and the id of a book, the book is added to the user, returns the user", response = User.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully updated user"),
+            @ApiResponse(code = 400, message = "Bad request"),
+            @ApiResponse(code = 404, message = "Resource not found"),
+            @ApiResponse(code = 500, message = "Internal server error")
+    })
     public User addBookToUser(@PathVariable Long userId, @PathVariable Long bookId) {
```
Comment (JaimLus) [inline src/main/java/wolox/training/controllers/UserController.java]: I would add @ApiParam() for the userId and bookId parameters.

File: src/main/java/wolox/training/controllers/UserController.java [hunk 6]
```diff
@@ -128,6 +181,13 @@ public User addBookToUser(@PathVariable Long userId, @PathVariable Long bookId)
...
      */
     @DeleteMapping("/{userId}/books/{bookId}")
+    @ApiOperation(value = "Given the id of a user and the id of a book, the book is deleted to the user, returns the user", response = User.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully updated user"),
+            @ApiResponse(code = 400, message = "Bad request"),
+            @ApiResponse(code = 404, message = "Resource not found"),
+            @ApiResponse(code = 500, message = "Internal server error")
+    })
     public User deleteBookToUser(@PathVariable Long userId, @PathVariable Long bookId) {
```
Comment (JaimLus) [inline src/main/java/wolox/training/controllers/UserController.java]: I would add @ApiParam() for the userId and bookId parameters.

**Review by JaimLus — COMMENTED** [review]
Nice job!

File: src/main/java/wolox/training/controllers/UserController.java [hunk 7]
```diff
@@ -37,6 +43,13 @@ public UserController(UserRepository userRepository, BookRepository bookReposito
      * @return {@link List<User>}
      */
     @GetMapping
+    @ApiOperation(value = "return all users", response = User[].class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Success"),
+            @ApiResponse(code = 401, message = "Not Authorized"),
+            @ApiResponse(code = 403, message = "Access forbidden"),
+            @ApiResponse(code = 404, message = "User Not Found")
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/controllers/UserController.java:51]: Remember that this method cannot return 404 :)

**Review by Jimenarosello — CHANGES_REQUESTED** [review]
nice job! Just remember to remove the 404 error from the list of possible status codes in the findAll method 😅

**Review by Jimenarosello — APPROVED** [review]

**Review by JaimLus — APPROVED** [review]
Nice job!!!!

**Pull request merged**